posthog/tasks/update_cache.py

```diff
@@ -46,26 +70,54 @@ def update_cached_items() -> None:
         .distinct("filters_hash")
     )
 
-    for item in items.filter(filters__isnull=False).exclude(filters={}).distinct("filters"):
+    for item in DashboardItem.objects.filter(
+        pk__in=Subquery(items.filter(filters__isnull=False).exclude(filters={}).distinct("filters").values("pk"))
+    ).order_by(F("last_refresh").asc(nulls_first=True))[0:PARALLEL_DASHBOARD_ITEM_CACHE]:
```

Comment:
This needs to be filtered by `refreshing=False` and maybe also `last_refresh__lte=timezone.now() - relativedelta(minutes=MAX_AGE_MINUTES)`